arch/risc-v/src/mpfs/mpfs_usb.c

```diff
@@ -3460,11 +3487,28 @@ static int mpfs_usb_interrupt(int irq, void *context, void *arg)
     {
       for (i = 0; i < MPFS_USB_NENDPOINTS; i++)
         {
+          /* Check if dead connections are back in business */
+
+          if (g_linkdead)
```

Comment:
Like the comment says, maybe it is worthwhile moving this global linkdead info to the ep (or use privep->linkdead here).

Reply:
The thing is that a TX ep has died, and we're checking the RX ep,  which is completely different EP. Not sure there's any relation to TX/RX eps; so to say if TX EP1 has died, activity on RX EP1 or EP2 might wake it up. Need to check though.

Reply:
Yes you are right, was not careful enough. I don't think anything in the USB standard mandates that the end point numbers for IN/OUT should be the same for the same connection. This comment can be ignored.

Reply:
Maybe it's possible to find out the pair, IN/OUT but not sure if it's visible in this driver level. I'll have a look, it'd be the best option, especially if there's multiple endpoints (IN, IN ,OUT OUT)..

Reply:
By tracing mpfs_allocep() I can see the eps being requested, but it's up to the higher level to pick the ones, so the driver doesn't see any IN/OUT pairs to rely on.

Reply:
For example, CDCACM has the following (configurable) epnos:

  devinfo.epno[CDCACM_EP_INTIN_IDX]   = CONFIG_CDCACM_EPINTIN;
  devinfo.epno[CDCACM_EP_BULKIN_IDX]  = CONFIG_CDCACM_EPBULKIN;
  devinfo.epno[CDCACM_EP_BULKOUT_IDX] = CONFIG_CDCACM_EPBULKOUT;